examples/build-eth-block/builder.sol

```diff
@@ -0,0 +1,201 @@
+        }
+
+        // Bubble sort, cause why not
+        uint256 n = bidsByEGP.length;
+        for (uint256 i = 0; i < n - 1; i++) {
+            for (uint256 j = i + 1; j < n; j++) {
+                if (bidsByEGP[i].egp < bidsByEGP[j].egp) {
+                    EgpRecordPair memory temp = bidsByEGP[i];
+                    bidsByEGP[i] = bidsByEGP[j];
+                    bidsByEGP[j] = temp;
+                }
+            }
+        }
+
+        Suave.DataId[] memory alldataIds = new Suave.DataId[](allRecords.length);
+        for (uint256 i = 0; i < bidsByEGP.length; i++) {
+            alldataIds[i] = bidsByEGP[i].dataId;
+        }
+
+        return buildAndEmit(blockArgs, blockHeight, alldataIds, "");
+    }
+
+    function buildAndEmit(
+        Suave.BuildBlockArgs memory blockArgs,
+        uint64 blockHeight,
+        Suave.DataId[] memory records,
+        string memory namespace
+    ) public virtual returns (bytes memory) {
+        require(Suave.isConfidential());
+
+        (Suave.DataRecord memory blockBid, bytes memory builderBid) =
+            this.doBuild(blockArgs, blockHeight, records, namespace);
+
+        emit BuilderBoostBidEvent(blockBid.id, builderBid);
+        emit DataRecordEvent(blockBid.id, blockBid.decryptionCondition, blockBid.allowedPeekers);
+        return bytes.concat(this.emitBuilderBidAndBid.selector, abi.encode(blockBid, builderBid));
+    }
+
+    function doBuild(
+        Suave.BuildBlockArgs memory blockArgs,
+        uint64 blockHeight,
+        Suave.DataId[] memory records,
+        string memory namespace
+    ) public returns (Suave.DataRecord memory, bytes memory) {
+        address[] memory allowedPeekers = new address[](2);
+        allowedPeekers[0] = address(this);
+        allowedPeekers[1] = Suave.BUILD_ETH_BLOCK;
+
+        Suave.DataRecord memory blockBid =
+            Suave.newDataRecord(blockHeight, allowedPeekers, allowedPeekers, "default:v0:mergedDataRecords");
+        Suave.confidentialStore(blockBid.id, "default:v0:mergedDataRecords", abi.encode(records));
+
+        (bytes memory builderBid, bytes memory payload) = Suave.buildEthBlock(blockArgs, blockBid.id, namespace);
+        Suave.confidentialStore(blockBid.id, "default:v0:builderPayload", payload); // only through this.unlock
+
+        return (blockBid, builderBid);
+    }
+
+    function emitBuilderBidAndBid(Suave.DataRecord memory dataRecord, bytes memory builderBid)
+        public
+        returns (Suave.DataRecord memory, bytes memory)
+    {
+        emit BuilderBoostBidEvent(dataRecord.id, builderBid);
+        emit DataRecordEvent(dataRecord.id, dataRecord.decryptionCondition, dataRecord.allowedPeekers);
+        return (dataRecord, builderBid);
+    }
+
+    function unlock(Suave.DataId dataId, bytes memory signedBlindedHeader) public returns (bytes memory) {
+        require(Suave.isConfidential());
+
+        // TODO: verify the header is correct
```

Comment:
do these still need to be added?

Reply:
I believe this is copied from the original example in suave-geth.  It's for a future PR :)